src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java

```diff
@@ -424,4 +490,30 @@ public boolean isInstantiable() {
         }
 
     }
+
+    private static class SlaveDisconnector extends MasterToSlaveCallable<Void, IOException> {
+
+        private static final long serialVersionUID = 8683427258340193283L;
+        
+		private static final Logger LOGGER = Logger.getLogger(SlaveDisconnector.class.getName());
+
+        @Override
+        public Void call() throws IOException {
+            Engine e = Engine.current();
+            // No engine, do nothing.
+            if (e == null) {
+                return null;
+            }
+            // Ensure the slave supports setNoReconnect
+            try {
+                Engine.class.getMethod("setNoReconnect", boolean.class);
+            } catch (NoSuchMethodException ex) {
+                return null;
+            }
+            e.setNoReconnect(true);
```

Comment:
`SlaveDisconnector` tells the slave pod's JNLP agent to not attempt any further reconnects. Without this, the slave JNLP agent will fail after the Jenkins build completes, leading to a final Pod phase of `Failed`.

Reply:
Again a code comment seems warranted here